### PR TITLE
feat(deps): remove dependency on `bee-inx`

### LINKED ISSUE #781: Move `bee-inx` into library part of Chronicle
Let's wait for https://github.com/iotaledger/iota.rs/pull/1292 to land first, then we can switch over.

### PR DESCRIPTION
**Pull request by grtlr**
## Linked Issues

* Closes #781

## Notes to Reviewer

As a reviewer, please pay particular attention to the following areas when reviewing this PR and tick the above boxes after you have completed the steps.

#### Test cases
* [x] Review and run tests that were added/changed.
* [x] Suggest places that may benefit from test cases.

#### INX Changes
* [x] Run chronicle using an INX connection.

#### Database Changes
* [x] Check for breaking changes in the data model and matching (conventional) commit message prefix.

### CODE REVIEW

File: src/inx/id.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+impl From<OutputId> for inx::proto::OutputId {
+    fn from(value: OutputId) -> Self {
+        Self {
+            id: [&value.transaction_id.0 as &[_], &value.index.to_le_bytes()].concat(),
+        }
+    }
+}
+
+// TODO: Should we write test cases for all the id types here?
```
Comment (Alex6323) [inline src/inx/id.rs:67]: (thanks for the low hanging fruit)
TODO
(should this instead be an issue?)

File: src/inx/ledger.rs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+#[allow(missing_docs)]
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct UnspentOutputMessage {
+    pub ledger_index: MilestoneIndex,
+    pub output: LedgerOutput,
+}
+
+#[allow(missing_docs)]
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Marker {
```
Comment (Alex6323) [inline src/inx/ledger.rs:26]: should this not be `MarkerMessage` for consistency? :trollface:

Comment (grtlr) [reply src/inx/ledger.rs:26]: Yes

File: src/bin/inx-chronicle/stardust_inx/error.rs [hunk 3]
```diff
@@ -5,7 +5,7 @@ use chronicle::types::tangle::MilestoneIndex;
 use thiserror::Error;
 
 #[derive(Debug, Error)]
-pub enum InxError {
+pub enum InxWriterError {
```
Comment (Alexandcoats) [inline src/bin/inx-chronicle/stardust_inx/error.rs:8]: Maybe `InxWorkerError`?

File: src/inx/id.rs (hunk 1 above)
Comment (grtlr) [reply src/inx/id.rs:67]: I opened an issue.

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**